s1m0nWasTaken/CS4337_BigDataProject
# Szymon/rating service

## Summary
Created RatingService where users can:

- Add ratings for a shop, ensuring a user cannot rate the same shop multiple times.
- Get all ratings for a specific shop.
- Update an existing rating.
- Delete a rating.

Implemented a check to prevent users from adding multiple ratings for the same shop.

## How to verify

- Use Postman to test the following endpoints:
- Add Rating (POST /ratings/add) - Ensure only one rating per user per shop is allowed.
- Get Ratings by Shop (GET /ratings/shop/{shopid}) - Retrieve all ratings for a specified shop.
- Update Rating (PUT /ratings/update/{id}) - Modify an existing rating.
- Delete Rating (DELETE /ratings/delete/{id}) - Remove an existing rating.
-
## Tests

- Added validation tests to verify users cannot add duplicate ratings.
- Tested CRUD operations for RatingService via Postman.

## Related
No related issues,
